src/vs/workbench/contrib/tasks/test/browser/taskTerminalStatus.test.ts

```diff
@@ -55,16 +55,16 @@ suite('Task Terminal Status', () => {
 		testTask = instantiationService.createInstance(TestTask);
 		problemCollector = instantiationService.createInstance(TestProblemCollector);
 	});
-	test('Should add failed status when there is an exit code on task end', async () => {
+	test('Should add failed status when there is an exit code on task end', () => {
 		taskTerminalStatus.addTerminal(testTask, testTerminal, problemCollector);
 		taskService.triggerStateChange({ kind: TaskEventKind.ProcessStarted });
 		assertStatus(testTerminal.statusList, ACTIVE_TASK_STATUS);
 		taskService.triggerStateChange({ kind: TaskEventKind.Inactive });
 		assertStatus(testTerminal.statusList, SUCCEEDED_TASK_STATUS);
 		taskService.triggerStateChange({ kind: TaskEventKind.End, exitCode: 2 });
-		setTimeout(() => assertStatus(testTerminal.statusList, FAILED_TASK_STATUS), 50);
+		poll<void>(() => Promise.resolve(), () => pollForStatus(testTerminal.statusList, FAILED_TASK_STATUS), 'status should be correct');
```

Comment:
You should add async back and await poll, otherwise poll will get called and the test will succeed before it's done

Reply:
the polling wasn't working, but instead going to just try to assert immediately and if that fails, then set timeout to assert 10ms later.

Reply:
if it still fails after 10ms, we should probably fix something in the code

Reply:
If it's 10ms it will almost certainly be flaky unfortunately, these will be running on CI agents which are typically low resources and containers/VMs on shared machines